src/neo/Consensus/ConsensusMessage.cs

```diff
@@ -15,7 +15,7 @@ public abstract class ConsensusMessage : ISerializable
         protected ConsensusMessage(ConsensusMessageType type)
         {
             if (!Enum.IsDefined(typeof(ConsensusMessageType), type))
-                throw new ArgumentOutOfRangeException(nameof(type));
+                throw new FormatException(nameof(type));
```

Comment:
This should throw `ArgumentOutOfRangeException` because it is checking the argument of this method.